oak/server/wasm_node.cc

```diff
@@ -514,4 +520,28 @@ wabt::interp::HostFunc::Callback WasmNode::OakChannelFind(wabt::interp::Environm
   };
 }
 
+wabt::interp::HostFunc::Callback WasmNode::OakRandomGet(wabt::interp::Environment* env) {
+  return [this, env](const wabt::interp::HostFunc* func, const wabt::interp::FuncSignature* sig,
+                     const wabt::interp::TypedValues& args, wabt::interp::TypedValues& results) {
+    LogHostFunctionCall(name_, func, args);
+
+    uint32_t offset = args[0].get_i32();
+    uint32_t size = args[1].get_i32();
+    if (!MemoryAvailable(env, offset, size)) {
+      LOG(WARNING) << "Node provided invalid memory offset+size";
+      results[0].set_i32(OakStatus::ERR_INVALID_ARGS);
+      return wabt::interp::Result::Ok;
+    }
+
+    std::uniform_int_distribution<uint8_t> distribution;
```

Comment:
Can you confirm that this does indeed obtain randomness from a trusted source when in Asylo? a link to something in their repo would be useful.

Reply:
Looking a bit deeper, the `std::default_random_engine` here looks like it falls through to `std::minstd_rand` (which is `std::linear_congruential_engine<std::uint_fast32_t, 48271, 0, 2147483647>`), so I've shifted to [`std::random_device`](https://en.cppreference.com/w/cpp/numeric/random/random_device).  That appears to eventually connect up to the `RDRAND` instruction; comment added to `wasm_node.h` with pointers.